wire-ios/Wire-iOS Tests/FilePreviewGeneratorTests.swift

```diff
@@ -23,31 +23,36 @@ import XCTest
 @testable import WireCommonComponents
 
 final class FilePreviewGeneratorTests: XCTestCase {
-    func testThatItDoesNotBreakOn0x0PDF() {
-        // given
-        let pdfPath = Bundle(for: type(of: self)).path(forResource: "0x0", ofType: "pdf")!
-        let sut = PDFFilePreviewGenerator(callbackQueue: OperationQueue.main, thumbnailSize: CGSize(width: 100, height: 100))
-        // when
-        let expectation = self.expectation(description: "Finished generating the preview")
-        sut.generatePreview(URL(fileURLWithPath: pdfPath), UTI: kUTTypePDF as String) { image in
-            XCTAssertNil(image)
-            expectation.fulfill()
+
+    private lazy var bundle = Bundle(for: Self.self)
+
+    func testThatItThrowsOn0x0PDF() async throws {
+
+        // Given
+        let pdfURL = try XCTUnwrap(bundle.url(forResource: "0x0", withExtension: "pdf"))
+        let sut = PDFFilePreviewGenerator(thumbnailSize: CGSize(width: 100, height: 100))
+
+        // When
+        do {
+            _ = try await sut.generatePreviewForFile(at: pdfURL)
+            XCTFail("Unexpected success")
+        } catch PDFFilePreviewGenerator.Error.failedToCreatePreview {
+            // Then
         }
-        // then
-        self.waitForExpectations(timeout: 2, handler: nil)
     }
 
-    func testThatItDoesNotBreakOnHugePDF() {
-        // given
-        let pdfPath = Bundle(for: type(of: self)).path(forResource: "huge", ofType: "pdf")!
-        let sut = PDFFilePreviewGenerator(callbackQueue: OperationQueue.main, thumbnailSize: CGSize(width: 100, height: 100))
-        // when
-        let expectation = self.expectation(description: "Finished generating the preview")
-        sut.generatePreview(URL(fileURLWithPath: pdfPath), UTI: kUTTypePDF as String) { image in
-            XCTAssertNil(image)
-            expectation.fulfill()
+    func testThatItThrowsOnHugePDF() async throws {
+
+        // Given
+        let pdfURL = try XCTUnwrap(bundle.url(forResource: "huge", withExtension: "pdf"))
+        let sut = PDFFilePreviewGenerator(thumbnailSize: CGSize(width: 100, height: 100))
+
+        // When
+        do {
+            _ = try await sut.generatePreviewForFile(at: pdfURL)
+            XCTFail("Unexpected success")
```

Comment:
I'm not sure I can follow you regarding this point, but generally I'm not a fan of little helpers and custom overrides of the SDK API. It creates confusion and makes the code less readable.

One example: the file Sequence+Async.swift
https://github.com/wireapp/wire-ios/pull/1284/files#diff-e04d4a1ce91dfd894ba34403a7d3106577fa4ced1180e137751208db506526ccL19

Maybe we're talking about different things.

Reply:
Sorry I'll clarify. I mean ideally we could test this using `XCTAssertThrowsError` but that doesn't work for `async` so it would be helpful to have a version of `XCTAssertThrowsError` that also works for async.

Reply:
@caldrian Feel free to close this comment. I wasn't asking for a change, just thinking out loud.